lms/djangoapps/mobile_api/decorators.py

```diff
@@ -37,6 +40,9 @@ def _wrapper(self, request, *args, **kwargs):
                         depth=depth,
                         check_if_enrolled=True,
                     )
+                except CourseAccessRedirect:
+                    # Raise a 404 if the user does not have course access
+                    raise Http404
                 except CoursewareAccessException as error:
                     return Response(data=error.to_json(), status=status.HTTP_404_NOT_FOUND)
```

Comment:
This is not your code, but same question about whether exposing the error with a 404 exposes what we are trying not to.